lib/fast_stats/means.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+# A class for managing the arithmetic and geometric means
+# of positive numbers
+module FastStats
+  class Means
+
+    attr_reader :means
+
+    def initialize
+      @means = {}
+    end
+
+    def add(name, val)
+      mean_for(name) << val
+    end
+
+    def summary(round: Mean::DEFAULT_ROUND)
+      means.values.reduce({}) do |summary, mean|
```

Comment:
Considering that you have activesupport, you might check out `each_with_object`, which might simplify this method if you go with the more deeply nested approach.

Reply:
@amiel with Tucler's suggestion I ended up being using `transform_values`